src/utils/helpers.js

```diff
@@ -203,30 +211,32 @@ const utils = {
     });
     return snakeCasedObj;
   },
-  containsPii(query) {
-    const piiRegex = Object.values(PII_REGEX);
+  containsPii(query, piiPattern) {
     const normalizedTerm = query.toLowerCase();
-
-    return piiRegex.some((regex) => regex.test(normalizedTerm));
+    return piiPattern.test(normalizedTerm);
   },
-  requestContainsPii(urlString) {
+  obfuscatePiiRequest(urlString) {
+    let obfuscatedUrl = urlString;
+
     try {
       const url = new URL(urlString);
       const paths = decodeURI(url?.pathname)?.split('/');
       const paramValues = decodeURIComponent(url?.search)?.split('&').map((param) => param?.split('=')?.[1]);
 
-      if (paths.some((path) => utils.containsPii(path))) {
-        return true;
-      }
+      PII_REGEX.forEach((regex) => {
+        paths.forEach((path) => {
+          if (utils.containsPii(path, regex.pattern)) obfuscatedUrl = obfuscatedUrl.replaceAll(path, regex.replaceBy);
```

Comment:
🤔 Wonder if there can be any unintended side effects of running replaceAll on the whole url. 

Can't think of any though. Anyone else see any issues with this?